### ISSUE TITLE
OnComplete - Supply Target

### ISSUE THREAD

**Issue by thematthopkins** [Demigiant/dotween#32]
My most common use case for OnComplete is to destroy the target after an animation.  Something like:  

```
mySpriteRenderer.DOFade(0f, 0.7f)
            .OnComplete(() => Object.DestroyObject(mySpriteRenderer.gameObject));
```

This works great, but from my understanding it's allocating memory for each closure because of the reference to mySprite.  Could OnComplete be overloaded to accept an Action<object> where object is the tween's target?

This would allow for the uglier, code below, but should avoid memory allocation:

```
mySpriteRenderer.DOFade(0f, 0.7f)
            .OnComplete(target => Object.DestroyObject(((MonoBehaviour)target).gameObject));
```

**Comment by alryazanov** [comment]
Hi!
There is my similar question:
Link: https://github.com/Demigiant/dotween/issues/14

**Comment by PedroMR** [comment]
Yes, I'd love to at the very least have the tween itself as a parameter. Doesn't seem like it would add much complexity or weight to the library while increasing usefulness while avoiding the closure performance hit.
